File: clemwek/reverse_integer.c
#include <stdio.h>

//  This is a program that takes an integer and returns a reversed integer

int reverse_integer(int x) {
    int reversed = 0;
    int remainder = 0;

    while (x != 0) {
        remainder = x % 10;
        reversed = reversed * 10 + remainder;
        x = x / 10;
    }
    return reversed;
}

int main() {
    printf("%d \n", reverse_integer(123));
    printf("%d \n", reverse_integer(500));
    printf("%d \n", reverse_integer(-90));
    printf("%d \n", reverse_integer(-56));
    printf("%d \n", reverse_integer(91));
}

File: clemwek/reverse_integer.py
# This is a program that takes an integer and returns a reversed integer

def reversed_integer(x):
    def reverse(x):
        x = str(x)
        # reverse the string and convert to integer
        x = x[::-1]
        return int(x)
    # Check if the the integer is negative 
    if x < 0:
        # take the absolute value and convert it to a string
        return -reverse(abs(x))
    else:
        return reverse(x)

# Test the function
print(reversed_integer(123)) # 321
print(reversed_integer(-123)) # -321
print(reversed_integer(500)) # 5
print(reversed_integer(-65)) # -56
print(reversed_integer(-90)) # -9
